reporting/fetch-reports.ts

```diff
@@ -0,0 +1,306 @@
+const loginButtonSelector =
+  "#login-form > .form-group > .controls > .signin > .btn";
+
+const numberOfReportsTotalSelector = "#meetingList span[name=totalRecords]";
+const addTrackingFieldToColumnsButtonSelector =
+  "#meetingList #trackfieldDropdownMenu > button";
+const addEventTrackingFieldColumnCheckboxSelector =
+  "#meetingList #trackfieldDropdownMenu label[alt=Event] > input[type=checkbox]";
+const nextPageLinkSelector =
+  "#meetingList > .list-col > .dynamo_pagination > li:nth-child(2) > a";
+
+const meetingListTableRowsSelector = "#meeting_list > tbody > tr";
+
+const modalExportWithMeetingDataSelector =
+  ".modal-dialog #contentDiv #withMeetingHeaderDiv input#withMeetingHeader";
+const modalExportButtonSelector =
+  ".modal-dialog #contentDiv button#btnExportParticipants";
+const modalMeetingInfoSelector = ".modal-dialog #contentDiv #meetingInfo";
+
+const COOKIES_PATH = "./cookies.json";
+
+const CONFIRM_VALUE = "i-have-edited-the-script-config-and-am-sure";
+
+const usage = makeScriptUsage({
+  description:
+    "Fetch zoom usage reports from the admin console using a headless Chrome browser via puppeteer",
+  usageParams: CONFIRM_VALUE,
+  exampleParams: CONFIRM_VALUE,
+});
+
+const [confirmationCheck] = process.argv.slice(2);
+if (confirmationCheck !== CONFIRM_VALUE) {
+  usage();
+}
+
+if (newLogin && username === "" && password === "") {
+  console.error("Error: username/password are required when newLogin=true.");
+  process.exit(1);
+}
+
+const keypress = async () => {
+  process.stdin.setRawMode(true);
+  return new Promise((resolve) =>
+    process.stdin.once("data", () => {
+      process.stdin.setRawMode(false);
+      resolve();
+    })
+  );
+};
+
+const makeHandleLogin = (page: Page) => async (
+  username: string,
+  password: string
+) => {
+  // Enter email address
+  await page.waitForSelector(loginEmailFieldSelector);
+  await page.type(loginEmailFieldSelector, username);
+
+  // Enter password
+  await page.type(loginPasswordFieldSelector, password);
+
+  // Click login
+  await page.waitForSelector(loginButtonSelector);
+  await page.click(loginButtonSelector);
+
+  // Note: the user may have to solve a captcha at this point, so don't timeout while they are doing so
+  await page.waitForNavigation({ timeout: 0 });
+};
+
+const makeGetNumberOfReportsTotal = (
+  page: Page
+) => async (): Promise<string> => {
+  await page.waitForSelector(numberOfReportsTotalSelector, {
+    visible: true,
+  });
+
+  const numberOfReportsTotal = await page.$eval(
+    numberOfReportsTotalSelector,
+    (el) => el.innerHTML
+  );
+  console.log(`Number of reports total: ${numberOfReportsTotal}`);
+
+  return numberOfReportsTotal;
+};
+
+// Log in to zoom, and download all participants reports from the above selected dates.
+(async () => {
+  const browser = await puppeteer.launch({ headless: false });
+  const page = await browser.newPage();
+
+  await page.setViewport({ width: 1400, height: 900 });
```

Comment:
Do we need to explicitly set the size of the viewport?

Reply:
That would have just been me reworking the existing code that was there, so.. not sure, but I’d rather leave it for now than spend any time looking into it.

I want to get rid of/change this method entirely when we can, to just use the APIs/similar.